src/lib/components/WellCompletions/utils/dataUtil.ts

```diff
@@ -0,0 +1,99 @@
+import { RangeMode, RangeModes, Well, Zone } from "../redux/types";
+export const preprocessData = (data: any) => {
+    return {
+        ...data,
+        wells: data.wells.map(well => {
+            let earliestCompDateIndex = Number.POSITIVE_INFINITY;
+            data.stratigraphy.forEach(zone => {
+                if (zone.name in well.completions) {
+                    //store earliest completion date
+                    const completion = well.completions[zone.name];
+                    const earliestDate = completion.t.find(
+                        (_, index) => completion.f[index] > 0
+                    );
+                    if (earliestDate !== undefined)
+                        earliestCompDateIndex = Math.min(
+                            earliestCompDateIndex,
+                            earliestDate
+                        );
+                }
+            });
+            return { ...well, earliestCompDateIndex };
+        }),
+    };
+};
+
+/**
+ * Util method to prepare stratigraphy and well data from the given time step range and other settings for plotting
+ * @param stratigraphy
+ * @param wells
+ * @param range
+ * @param rangeDisplayMode
+ * @param hideZeroCompletions
+ * @returns
```

Comment:
Looks like an incomplete documentation? I do actually prefer the documentation style you used further up with the comments in the interface definition.

Reply:
further up with the comments in the interface definition?